graphql_compiler/compiler/expressions.py

```diff
@@ -576,6 +624,10 @@ def to_gremlin(self):
         raise AssertionError(u'FoldedContextField are not used during the query emission process '
                              u'in Gremlin, so this is a bug. This function should not be called.')
 
+    def to_cypher(self):
+        """Not implemented yet."""
+        raise NotImplementedError()
```

Comment:
Do we want to implement this (FoldedContextField) in this PR, since we're implementing to_cypher methods for all the Expression objects here? Same question for the to_cypher methods in FoldCountContextField and UnaryTransformation. 

If it makes sense to do them in this PR rather than doing it in a separate PR, I can look into implementing those methods right now.

Reply:
Doing this well would probably require implementing `@fold` in general, otherwise there's no possible way to test that we've implemented it correctly. I think that would be too big of a lift for this PR.